### PR TITLE
Cleaner errors and optionally enable skipping wrong packages

### PR DESCRIPTION
**Pull request by htgoebel**
This brings in https://github.com/dkg/python-pgpdump/tree/cleaner-errors plus some minor changes by me.

### CODE REVIEW

**Comment by htgoebel** [comment]
Please note: This pull-request also includs some changes already in #3 due to how dkg did apply the changes in his repo. Anyway, please merge #2 first to have that branch merge into here. Thus others can easily see that dpk'g changes are in here already. Thanks.

**Comment by alex-nitrokey** [comment]
Thank you very much for your effort creating this PR! I will look into this one during the week.

File: pgpdump/utils.py [hunk 1]
```diff
@@ -71,27 +73,93 @@ def crc24(data):
...
 def get_int8(data, offset):
     '''Pull eight bytes from data at offset and return as an integer.'''
+    if offset+8 > len(data):
+        raise PgpdumpException(
+            "Wants an 8-byte integer at position %d of packet of size %d"
+            % (offset, len(data)))
     return (get_int4(data, offset) << 32) + get_int4(data, offset + 4)
 
 
+def encode_packet(tag, new, data, armored=False):
```
Comment (alex-nitrokey) [inline pgpdump/utils.py:102]: It would be great to have a function description here.

**Review by alex-nitrokey — CHANGES_REQUESTED** [review]
Thanks again for this PR. Would you mind creating a test case in test.py that checks and demonstrates the new feature of packet skipping and encoding? That'd be awesome. I guess, I would merge this PR anyway, but I would really prefer having this tested in future to prevent breaking stuff unintentionally.

**Comment by htgoebel** [comment]
@alex-nitrokey Sorry for the delay, I was completely distracted from this issue. Now I updates the pull-request and added what you requested.

**Comment by htgoebel** [comment]
Lint errors are from code untouched by me.

**Comment by alex-nitrokey** [comment]
> Sorry for the delay, I was completely distracted from this issue. Now I updates the pull-request and added what you requested.

No harm done, thanks for your contribution and efforts!

> Lint errors are from code untouched by me.

Some are ;) I think I fixed them, but couldn't apply them directly to you PR, so I created a [new branch](https://github.com/alex-nitrokey/python-pgpdump/tree/lint-fixes)... See [this commit](https://github.com/alex-nitrokey/python-pgpdump/commit/b3b858a9fde21d2443e02401e51cbadec9b4572c). If you check the correct box in the PR settings I might be able to include it right away here. Or please see if you can just merge the commit. Or I will just merge your PR and fix the minor lint errors afterwards.

Anyway I thank you for contributing to this software I am not even using anymore...

**Comment by alex-nitrokey** [comment]
Again, thanks a lot!

**Pull request merged**